### PR TITLE
refactor: remove hard pins in requirements-common.txt

### PR DESCRIPTION
**Pull request by dkunitsk**
Signed-off-by: Dmitriy Kunitskiy <dkunitskiy@lyft.com>

### Summary of Changes
The current setup introduces hard pins for `amundsen-common` and the hard pins may also be inconvenient for custom setups of the amundsen microservices. We are attempting to roll forward with this removal 
After discussing with @szczeles and @feng-tao, we've decided to try removing hard pins completely and see what kind of trouble we run into. 

I've also deleted some dependencies which appear to no longer be used. 

### Tests

Rebuilt containers (following https://github.com/amundsen-io/amundsen/blob/master/docs/developer_guide.md#building-local-changes) and ran the quick start guide with sample data. Don't see any issues so far! 

### CheckList
Make sure you have checked **all** steps below to ensure a timely review.
- [ ] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
    - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [ ] PR includes a summary of changes.
- [ ] PR adds unit tests, updates existing unit tests, __OR__ documents why no test additions or modifications are needed.
- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
    - All the public functions and the classes in the PR contain docstrings that explain what it does

### CODE REVIEW

**Comment by feng-tao** [comment]
make sense

**Review by feng-tao — APPROVED** [review]
+1, i wonder whether you could do a quick sanity test and see if the demo services are still up and running with dep upgrade

File: requirements-common.txt [hunk 1]
```diff
@@ -5,26 +5,26 @@
 
 # It is recommended to always pin the exact version (not range) - otherwise common upgrade won't trigger unit tests
```
Comment (dorianj) [inline requirements-common.txt:6]: Remove this comment if no longer true?

Comment (dkunitsk) [reply requirements-common.txt:6]: ha, thank you

**Comment by dkunitsk** [comment]
@feng-tao did that, no issues afaict.

**Pull request merged**